templates/README.md

```diff
@@ -0,0 +1,276 @@
+
+## Example file
+
+### template.yaml
+
+```YAML
+# vim: sts=2 sw=2 et
+apiVersion: v1
+kind: Template # Openshift kind of template or something "better"
+metadata:
+  name: windows-10
+  annotations:
+    openshift.io/display-name: "Generic Microsoft Windows 7+ template"
+    description: "Basic windows template long description with details"
+    openshift.io/long-description: >-
+      Long description of the template
+    openshift.io/provider-display-name: "Red Hat, Inc."
+    openshift.io/documentation-url: "https://cnv.io/..."
+    openshift.io/support-url: "https://access.redhat.com"
+    iconClass: icon-windows    
+
+    # The `defaults` set of annotations is meant as a hint only
+    # and is not going to be processed by the stock openshift templating
+    # mechanism. CNV UI can and should use it though.
+    # The information encoded there informs the tooling (UI)
+    # about names of devices that are to be used as templates
+    # for adding additional disk, networks, volumes and such
+
+    # The goal of default disk is to define what kind of disk
+    # is supported by the OS mainly in terms of bus (ide, scsi,
+    # sata, virtio, ...)
+    defaults.template.cnv.io/disk: default-disk
+
+    # The goal of default volume is to be able to configure mostly
+    # performance parameters like caches if those are exposed
+    # by the underlying volume implementation.
+    defaults.template.cnv.io/volume: default-volume
+
+    # The goal of default network is similar to default-disk
+    # and should be used as a template to ensure OS compatibility
+    # and performance
+    defaults.template.cnv.io/nic: default-nic
+
+    # The goal of default network is similar to default-volume
+    # and should be used as a template that specifies performance
+    # and connection parameters (L2 bridge for example)
+    defaults.template.cnv.io/network: default-network
+
+    # Replacements useful for finishing the transformation
+    # This is an extension we could use for specifying common
+    # fields for all disks/nics. Requires extra implementation
+    # that will do the find and replace using a syntax documented
+    # here: https://kubernetes.io/docs/reference/kubectl/jsonpath/
+    # The jsonpath root is the objects: element of the template
+    # XXX There is currently an issue with annotations not supporting
+    #     list values
+    template.cnv.io/replace:
+    - path: "/VM/disks/*/type"
+      value: virtio
```

Comment:
This particular example is redundant to the default disk bus. Is there another example that could be used?